nselib/http.lua

```diff
@@ -1252,6 +1252,45 @@ local function request(host, port, data, options)
   return response
 end
 
+--- This function merges the cookies received in <code>response.cookies</code> 
+-- to the cookies that already exist in the options. 
+--
+-- The merge is based on RFC 6265 and when a different cookie with same <code>
+-- name</code>, <code>path</code> and <code>domain</code> is received, it replaces
+-- the old cookie, else it gets appended at the end of <code>options.cookies</code table.
+
+-- @param response The response received from the server
+-- @param options The options table having previously received cookies.
+-- @return A response table
+-- @return An options table, with the cookies from the response table appended.
+-- @see get
+local function merge_cookie_table(response, options)
+  local flag = false
+  for r_index,r_cookie in pairs(response.cookies) do
+    for o_index,o_cookie in pairs(options.cookies) do
+      flag = false
+      if(r_cookie.name == o_cookie.name) then
+        --We need to check if domain and path are equal.
+        --Note:If both domain and path are nil for r_cookie and o_cookie,
+        --we need to change the cookie value 
+        --Nmap http library doesnt take cookie.domain into consideration anywhere.
+        --See RFC 6265 Section 5.3 for how duplicate cookies are handled
+        if(r_cookie.domain == o_cookie.domain and r_cookie.path == o_cookie.path) then 
+          --If options.no_cookie_overwrite is set, we simply ignore this cookie
+          if(options.no_cookie_overwrite == true ) then break end 
```

Comment:
If we don't want the library to update the cookie jar then this check shouldn't be inside the loop. We don't need to iterate through the cookies if we are not planning on updating them.